lib/rouge/lexers/toml.rb

```diff
@@ -28,9 +28,14 @@ class TOML < RegexLexer
 
         rule %r/\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}Z/, Literal::Date
 
-        rule %r/(\d+\.\d*|\d*\.\d+)([eE][+-]?[0-9]+)?j?/, Num::Float
-        rule %r/\d+[eE][+-]?[0-9]+j?/, Num::Float
-        rule %r/\-?\d+/, Num::Integer
+        rule %r/[+-]?\d+(_\d+)*\.\d+(_\d+)*([eE][+-]?\d+(_\d+)*)?/, Num::Float
+        rule %r/[+-]?\d+(_\d+)*[eE][+-]?\d+(_\d+)*/, Num::Float
+        rule %r/[+-]?(nan|inf)/, Num::Float
+
+        rule %r/0x\h+(_\h+)*/, Num::Integer
+        rule %r/0o[0-7]+(_[0-7]+)*/, Num::Integer
```

Comment:
**suggestion:** We can refine this token to be `Num::Oct`. For most themes, it will be highlighted the same as `Number` but give flexibility for other sophisticated themes. Additionally, since no capture is required we can mark it as a non-captured group with `?:`.

```suggestion
        rule %r/0o[0-7]+(?:_[0-7]+)*/, Num::Oct
```